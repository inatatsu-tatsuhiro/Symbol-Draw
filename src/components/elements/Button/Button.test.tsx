import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Component from './index'

const text = 'BUTTON TEXT'

test('button: matching text', () => {
  const { container } = render(
    <Component text={text} onClick={() => console.log('')} />
  )
  expect(container.innerHTML).toMatch(text)
})

test('button: snapshot testing', () => {
  const { asFragment } = render(
    <Component text={text} onClick={() => console.log('')} />
  )
  expect(asFragment()).toMatchSnapshot()
})

test('button: onClick', () => {
  const handleClick = jest.fn()
  const { getByText } = render(
    <Component text={text} onClick={() => handleClick()} />
  )
  fireEvent.click(getByText(text))
  expect(handleClick).toHaveBeenCalledTimes(1)
})
