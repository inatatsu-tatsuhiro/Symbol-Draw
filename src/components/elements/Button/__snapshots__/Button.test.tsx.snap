// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button: snapshot testing 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root sc-gsDKAQ jtLrrK css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    BUTTON TEXT
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
